atom/browser/api/atom_api_web_contents.cc

```diff
@@ -1444,12 +1444,17 @@ v8::Local<v8::Promise> WebContents::HasServiceWorker() {
   auto* context = GetServiceWorkerContext(web_contents());
   if (!context) {
     promise->RejectWithErrorMessage("Unable to get ServiceWorker context.");
+    return promise->GetHandle();
+  }
+
+  GURL url = web_contents()->GetLastCommittedURL();
+  if (!url.is_valid()) {
```

Comment:
When would a committed url be invalid ? Its a virtual url generated by the navigation controller. The DCHECK you hit is probably the other url that should be encompassed isn't recognized, in which case try using visible URL. Also using the result of committed URL to test page loading state is not a good idea.